File: backend/schemas/penalty.py
from datetime import datetime
from typing import Optional
from pydantic import BaseModel, Field


class PenaltyCreate(BaseModel):
    offender_id: str
    reason: str
    penalty_type: str = "money"
    amount: Optional[float] = Field(None, ge=0)
    action: Optional[str] = None
    photo_url: Optional[str] = None
    note: Optional[str] = None


class PenaltyOut(BaseModel):
    id: str
    issuer_id: str
    offender_id: str
    reason: str
    penalty_type: str
    amount: Optional[float]
    action: Optional[str]
    photo_url: Optional[str]
    note: Optional[str]
    is_done: bool
    done_at: Optional[datetime]
    created_at: datetime

    class Config:
        from_attributes = True

File: frontend/miniapp/src/api/penalty.ts
import { request } from '@/utils/request'

export interface Penalty {
  id: string
  issuer_id: string
  offender_id: string
  reason: string
  penalty_type: 'money' | 'action'
  amount: number | null
  action: string | null
  photo_url: string | null
  note: string | null
  is_done: boolean
  done_at: string | null
  created_at: string
}

export interface PenaltyCreate {
  offender_id: string
  reason: string
  penalty_type?: 'money' | 'action'
  amount?: number
  action?: string
  photo_url?: string
  note?: string
}

export const penaltyApi = {
  list: () =>
    request<Penalty[]>({ url: '/api/penalties' }),

  create: (data: PenaltyCreate) =>
    request<Penalty>({ url: '/api/penalties', method: 'POST', data }),

  done: (id: string) =>
    request<Penalty>({ url: `/api/penalties/${id}/done`, method: 'POST' }),

  delete: (id: string) =>
    request<{ message: string }>({ url: `/api/penalties/${id}`, method: 'DELETE' }),
}
